---
title: "kidney-tumor"
author: "yincy"
date: "5/24/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# kidney primary tumor
```{r, eval=FALSE}
library(TCGAbiolinks)
library(magrittr)
library(dplyr)
library(SummarizedExperiment)
library(ggplot2)

kidney_tumor_projects <- c("TARGET-WT", "TCGA-KIRP", "TCGA-KIRC", "TCGA-KICH", "TARGET-CCSK")

ccrcc_summary <- getProjectSummary(project = "TCGA-KIRC", legacy = F)
ccrcc_sample_summary <- getSampleFilesSummary(project = "TCGA-KIRC", legacy = F)

ccrcc_query_info <- GDCquery(project = "TCGA-KIRC", # ccrCC
                         data.category = "Transcriptome Profiling", 
                         experimental.strategy = "RNA-Seq", 
                         data.type = "Gene Expression Quantification", 
                         sample.type = "Primary Tumor", 
                         workflow.type = "STAR - Counts",
                         access = "open", 
                         legacy = F)

saveRDS(ccrcc_query_info, "/home/yincy/git/Data/Kidney/TCGA/ccRCC/ccrcc_query_info.rds")
getResults(query = ccrcc_query_info) 

ccrcc_query_info <- readRDS("/home/yincy/git/Data/Kidney/TCGA/ccRCC/ccrcc_query_info.rds")

GDCdownload(query = ccrcc, 
            directory = "/home/yincy/git/Data/Kidney/TCGA/ccRCC", 
            files.per.chunk = 10, 
            method = "api")

ccrcc <- GDCprepare(query = ccrcc_query_info, 
                    directory = "/home/yincy/git/Data/Kidney/TCGA/ccRCC")

saveRDS(ccrcc_counts, 
        file = "/home/yincy/git/Data/Kidney/TCGA/ccRCC/ccrcc_summrizedexpriment.rds")

ccrcc <- readRDS("/home/yincy/git/Data/Kidney/TCGA/ccRCC/ccrcc_summrizedexpriment.rds")

colData(ccrcc)[, 1:10, 1:3]

ccrcc %>% colData()
ccrcc %>% rowData() %>% as.data.frame() %>% filter(gene_name == "PDGFB")
assayNames(ccrcc)

sample_info <- colData(ccrcc) %>% as.data.frame()
pdgfb_fpkm <- assay(ccrcc, "fpkm_unstrand")["ENSG00000100311.17", ] %>% 
    as.data.frame() %>% tibble::rownames_to_column() %>% 
    magrittr::set_colnames(c("barcode", "fpkm_pdgfb"))

pdgfb_fpkm <- pdgfb_fpkm %>% left_join(sample_info, by = c("barcode" = "barcode"))

pdgfb_summary <- pdgfb_fpkm %>% 
    filter(!is.na(ajcc_pathologic_stage)) %>% 
    group_by(ajcc_pathologic_stage) %>% 
    summarise(n = n(), 
              mean = median(fpkm_pdgfb), 
              se = sd(fpkm_pdgfb)/sqrt(n - 1))
seg_df <- data.frame(
    x = rep(1, 3), 
    y = max(pdgfb_fpkm$fpkm_pdgfb) * c(1.1, 1.2, 1.3) + 0.9, 
    xend = seq(2, 4, 1), 
    yend = max(pdgfb_fpkm$fpkm_pdgfb) * c(1.1, 1.2, 1.3) + 0.9
)

text_df <- data.frame(
    x = c(1.5, 2.0, 2.5), 
    y = max(pdgfb_fpkm$fpkm_pdgfb) * c(1.08, 1.18, 1.28) + 1.02,
    label = c("*", "*", "**")
)

pdgfb_fpkm %>% 
    filter(!is.na(ajcc_pathologic_stage)) %>% 
    ggplot(aes(ajcc_pathologic_stage, fpkm_pdgfb)) +
    geom_boxplot(outlier.alpha = 0) +
    geom_line(data = pdgfb_summary, aes(ajcc_pathologic_stage, mean), group = 1, col = "gray30", size = 1.2) +
    geom_jitter(width = 0.2) +
    geom_segment(data = seg_df, 
                 aes(x = x, y = y, xend = xend, yend = yend)) +
    geom_text(data = text_df, 
              aes(x = x, y = y, label = label), size = 8) +
    scale_x_discrete(name = "AJCC pathologic stage", 
                     labels = paste(pdgfb_summary$ajcc_pathologic_stage, " \nn = ", pdgfb_summary$n, sep = "")) +
    scale_y_continuous(name = "log2( FPKM of PDGFB )", 
                       trans = "log2") +
    theme_classic() +
    labs(title = "expression of PDGFB in different AJCC pathologic stage", 
         subtitle = "data from TCGA-KIRC (ccRCC) project")
ggsave("/home/yincy/git/Data/Kidney/TCGA/Pdgfb-ccRCC-stage-expression.tiff", 
       width = 16.18, height = 10)
```

```{r}
library(multcomp)

fit <- aov(fpkm_pdgfb ~ ajcc_pathologic_stage, data = pdgfb_fpkm) 
pairwise <- TukeyHSD(fit)
```


```{r}
barcodes <- colData(ccrcc)$patient %>% unique()
clinical_query_info <- GDCquery(project = "TCGA-KIRC", 
                          data.category = "Clinical", 
                          file.type = 'xml', 
                          barcode = barcodes)

saveRDS(clinical_info, file = "/home/yincy/git/Data/Kidney/TCGA/ccRCC/clinical_query_info.rds")

GDCdownload(query = clinical_query_info, 
            directory = "/home/yincy/git/Data/Kidney/TCGA/ccRCC/", 
            method = "api", 
            files.per.chunk = 10)


clinical <- GDCprepare_clinic(query = clinical_query_info, 
                              clinical.info = "patient", 
                              directory = "/home/yincy/git/Data/Kidney/TCGA/ccRCC/")
saveRDS(clinical, 
        file = "/home/yincy/git/Data/Kidney/TCGA/ccRCC/clinical_info.rds")
clinical %>% str
```





